raiden/network/transport/matrix/transport.py

```diff
@@ -443,10 +451,12 @@ def stop(self):
             if retrier:
                 retrier.notify()
 
+        self._address_mgr.stop()
+        self._client.stop_listener_thread()  # stop sync_thread, wait on client's greenlets
+
         self._client.set_presence_state(UserPresence.OFFLINE.value)
 
-        self._client.stop_listener_thread()  # stop sync_thread, wait client's greenlets
-        # wait own greenlets, no need to get on them, exceptions should be raised in _run()
+        # wait on own greenlets, no need to get on them, exceptions should be raised in _run()
         gevent.wait(self.greenlets + [r.greenlet for r in self._address_to_retrier.values()])
```

Comment:
`self._client.set_presence_state(UserPresence.OFFLINE.value) `

before the `gevent.wait(self.greenlets)` might lead to the condition where we set our presence back to online again right? There might be greenlets around from `_maybe_invite` etc. interacting with the server even though the listeners are stopped.

Reply:
Yes, you're right. Good catch. I'll flip the order.